client/src/assets/game_data/gameData.ts

```diff
@@ -6,7 +6,11 @@ import { characters as charactersList } from "./characterData";
 const characters: Record<Role, Character> = Object.fromEntries(
   charactersList.map((role) => [
     role.id as Role,
-    { ...role, id: role.id as Role },
+    {
+      ...role,
+      id: role.id as Role,
+      imageSrc: new URL(`../icon/role/${role.imageSrc}`, import.meta.url).href,
```

Comment:
I am not confident this will work, we don't know the url at build time.  Instead, we should pull imageSrc out and add it to a separate map to get the URL.  We may be able to do this via dynamic imports, but could also just write a script that generates an index file

Reply:
And then we just make sure all the pngs match the ids we use

Reply:
Yeah I took a while figuring URL resolution out.  Haven't learned enough yet, but this seemed promising:
[https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url](https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url)